### PR TITLE
[ASSETS-14816] Call `onEnd` when closing connection

### PR DESCRIPTION
**Pull request by tmathern**
## Description

- Remove an `async` function that is not async
- Call `onEnd` when closing connection
- Up dependencies

Fixes # [ASSETS-14816](https://jira.corp.adobe.com/browse/ASSETS-14816)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by badvision — APPROVED** [review]
Way to go!!  This was a doozy of a problem.  I will be very interested to see how our stats are affected by this improvement.

File: lib/probe/http-client.js [hunk 1]
```diff
@@ -268,6 +269,11 @@ class HttpRequestMetrics {
             this.onEnd();
         });
 
+        response.on("close", () => {
+            debug("⚡️ [response] close");
+            this.onEnd();
```
Comment (jdelbick) [inline lib/probe/http-client.js:274]: so this was the bug?
some http requests just "close" not "end"?

**Review by jdelbick — APPROVED** [review]
LGTM, one question

File: lib/newrelic.js [hunk 2]
```diff
@@ -113,7 +113,7 @@ class NewRelic {
     /**
      * Call this when the action activation finishes successfully.
      */
-    async activationFinished() {
+    activationFinished() {
```
Comment (jdelbick) [inline lib/newrelic.js:116]: nice to have: can you remove all the `await`'s in the tests for these? https://github.com/adobe/node-openwhisk-newrelic/blob/50fbd58c5b2c9e054dac145b36d1311544626b2b/test/newrelic.test.js#L165

**Review by badvision — APPROVED** [review]

File: lib/newrelic.js (hunk 2 above)
Comment (jdelbick) [reply lib/newrelic.js:116]: also we call `await` on this in commons:
https://github.com/adobe/asset-compute-commons/blob/master/lib/wrap.js#L70
.what this method supported too be async but we didn't return on: 
```
return clearTimeout(this.actionTimeoutHandlerId);
```

Comment (tmathern) [reply lib/newrelic.js:116]: One library at a time, @jdelbick. Let me start releasing this. 
Also await/async is "only" syntactic sugar, await a sync function shouldn't interfere.

**Pull request merged**